### DISCUSSION TITLE
No web interface

### DISCUSSION THREAD

**Discussion by jcherney** [jgyates/genmon#1116] [Questions]
Hi all

Just swapped out my PI3 for a new one. Not getting the same error emails as before so that's good. 

Still can't reach the web interface. And I've tried everything that I can think of. The port forward seems to be configured properly but that and $4 will get me a latte at Starbucks! But I digress..

As I am not at all versed in the coding language, can someone guide me to updating the Genmon software thru the SSH access (which I can do)?

Thanks.

**Comment by jgyates** [comment]
Can you elaborate on what you mean by "port forwarding seems to be configured properly"? If you have port forwarding enabled on your router, then this is probably you problem. You should not be exposing genmon to the internet outside of your router. Not only will this result in your [genmon installation getting hacked by port sniffing software](https://github.com/jgyates/genmon/issues/702), it could result in your router blocking access.

Here are similar issues:

https://github.com/jgyates/genmon/issues?q=label%3Asecurity

https://github.com/jgyates/genmon/issues?q=+label%3ARouter+

**Comment by mac1-131** [comment]
Let's just say you really want to expose it. I do (and depend on TFA to not get hacked).

Any chance your router changed your IP address with the new pi which has a different MAC address? That will mess up your port forwarding rule.

Every router is different, but you need to get your pi on a static ip, which these days is better set up in the router than in the pi. Leave the pi on DHCP, and set the router to assign a static IP to the pi which matches your port forwarding rule.

But be sure you understand the ramifications and don't do it unless you have a TFA authenticator working or you will eventually get hacked and have hackers operating your generator.

**Reply by jcherney** [reply]
I removed the port forwarding. I was under the impression it was needed- and that was debunked quickly.!
Thanks!

**Reply by jgyates** [reply]
not getting your credentials compromised is only have of the problem. The web server that genmon uses (Flask) is not designed for a high traffic environment so this makes the web server more prone to denial of service attacks (i.e. a bot will continue to probe your site and fail to log in but it will not allow anyone to log in, even you, as the server is overloaded or crashed). I am not sure how likely this is but I do know that the Flask project warns of this.
